src/flag_gems/ops/conv2d.py

```diff
@@ -0,0 +1,311 @@
+        conv2d_forward_config(64, 128, 128, n_warps=4, num_stages=4),
+        conv2d_forward_config(64, 128, 64, n_warps=2, num_stages=4),
+        conv2d_forward_config(64, 64, 64, n_warps=2, num_stages=4),
+        conv2d_forward_config(64, 32, 32, n_warps=2, num_stages=4),
+        conv2d_forward_config(64, 16, 32, n_warps=2, num_stages=4),
+        conv2d_forward_config(64, 16, 16, n_warps=2, num_stages=4),
+        conv2d_forward_config(32, 32, 32, n_warps=2, num_stages=4),
+        conv2d_forward_config(32, 16, 32, n_warps=2, num_stages=4),
+        conv2d_forward_config(32, 16, 16, n_warps=2, num_stages=4),
+    ],
+    key=[
+        "in_n",
+        "kernel_c",
+        "in_height",
+        "in_width",
+        "out_c",
+        "out_height",
+        "out_width",
+        "kernel_height",
+        "kernel_width",
+        "stride_height",
+        "stride_width",
+        "padding_height",
+        "padding_width",
+        "groups",
+    ],
+)
+@triton.jit
+def conv2d_forward_kernel(
+    input_pointer,
+    weight_pointer,
+    output_pointer,
+    in_n,
+    in_height,
+    in_width,
+    out_c,
+    out_height,
+    out_width,
+    input_n_stride,
+    input_c_stride,
+    input_height_stride,
+    input_width_stride,
+    weight_n_stride,
+    weight_c_stride,
+    weight_height_stride,
+    weight_width_stride,
+    output_n_stride,
+    output_c_stride,
+    output_height_stride,
+    output_width_stride,
+    kernel_c: tl.constexpr,
+    kernel_height: tl.constexpr,
+    kernel_width: tl.constexpr,
+    stride_height: tl.constexpr,
+    stride_width: tl.constexpr,
+    padding_height: tl.constexpr,
+    padding_width: tl.constexpr,
+    groups: tl.constexpr,
+    BLOCK_NI_HO_WO: tl.constexpr,
+    BLOCK_CI: tl.constexpr,
+    BLOCK_CO: tl.constexpr,
+):
+    pid_ni_ho_wo = tl.program_id(0)
+    pid_co = tl.program_id(1)
+    pid_group = tl.program_id(2)
+
+    # caculate in_n out_height out_weight value in kernel
+    ni_ho_wo_offset = pid_ni_ho_wo * BLOCK_NI_HO_WO + tl.arange(0, BLOCK_NI_HO_WO)
+    ni_ho_offset = ni_ho_wo_offset // out_width
+    in_n_point_value = ni_ho_offset // out_height
+    output_height_point_value = ni_ho_offset % out_height
+    output_width_point_value = ni_ho_wo_offset % out_width
+
+    # Load the input and weight pointers. input and weight are of shape
+    # [in_n, groups, in_c, in_height, in_width] and [groups, out_c, in_c, kernel_height, kernel_width]
+    out_per_group_c = out_c // groups
+    output_c_offset = pid_co * BLOCK_CO + tl.arange(0, BLOCK_CO)
+    input_pointer += (
+        input_n_stride * in_n_point_value + input_c_stride * pid_group * kernel_c
+    )[:, None]
+    weight_pointer += (
+        weight_n_stride * output_c_offset
+        + weight_n_stride * pid_group * out_per_group_c
+    )[None, :]
+
+    accum = tl.zeros((BLOCK_NI_HO_WO, BLOCK_CO), dtype=tl.float32)
+
+    for h in range(kernel_height):
+        for w in range(kernel_width):
```

Comment:
since kernel_height and kernel_width are processed as loop iterator range, why not support those non-power-of-2?